File: src-ts/graphs/Edge.ts
/*
 * Edge is a set of two vertices that are connected by the edge.
 *
 */
import { Vertex } from "./Vertex";

interface IEdge {
    source: Vertex;
    target: Vertex;
    weight: number;
}

interface IDirectedEdge extends IEdge {
    
}

export class Edge {

    /**
     * This is an array of the two vertices on which the edge is incident.
     *
     * The vertices should be represented by their id: string properties.
     */
    connects: Array<string>;

    constructor([v1, v2]) {
        this.connects = [v1, v2];
    }
}

File: src/eu/libal/intrographs/edge/Edge.java
package eu.libal.intrographs.edge;

import eu.libal.intrographs.vertex.IVertex;

public class Edge<VertexType> implements IEdge {

    private IVertex<VertexType> source;
    private IVertex<VertexType> target;

    public Edge(IVertex<VertexType> source, IVertex<VertexType> target)
    {
        this.source = source;
        this.target = target;
    }

    @Override
    public IVertex getTarget() {
        return target;
    }

    @Override
    public IVertex getSource() {
        return source;
    }
}
